mlflow/server/handlers.py

```diff
@@ -3784,6 +3796,209 @@ def _get_dataset_records_handler(dataset_id):
+                try:
+                    scorer = get_builtin_scorer_by_name(scorer_name)
+                    scorer_instances.append(scorer)
+                except Exception as e:
+                    raise MlflowException(
+                        f"Failed to create scorer '{scorer_name}': {e}",
+                        INVALID_PARAMETER_VALUE,
+                    )
+            
+            if not scorer_instances:
+                raise MlflowException(
+                    "No valid scorers provided for optimization",
+                    INVALID_PARAMETER_VALUE,
+                )
+
+            # Log optimization parameters
+            _logger.info(f"Starting prompt optimization job: {job_id}")
+
+            # Set up LLM parameters
+            
+            # Parse target_llm to create LLMParams
+            target_llm = job['target_llm']
+            llm_params = LLMParams(model_name=target_llm)
+            
+            # Set up optimizer config
+            algorithm = job['algorithm']
+            optimizer_config = OptimizerConfig(
+                algorithm=algorithm,
+                autolog=True,
+            )
+
+            prompt_input = load_prompt(job["prompt_url"])
+            
+            # Call the actual optimize_prompt function
+            result = optimize_prompt(
+                target_llm_params=llm_params,
+                prompt=prompt_input,
+                train_data=train_df,
+                scorers=scorer_instances,
+                eval_data=eval_df,
+                optimizer_config=optimizer_config,
+            )
+            
+            _logger.info(f"Prompt optimization job {job_id} completed.")
+
+            # Save optimization result
+            job["result"] = {
+                "prompt_url": result.prompt.uri,
+                "evaluation_score": result.final_eval_score or 0.0
+            }
+            job["status"] = "COMPLETED"
+
+        except Exception as e:
+            job["status"] = "FAILED"
+            job["error"] = str(e)
+            _logger.error(f"Prompt optimization job {job_id} failed: {e}")
+
+
+# Global job manager instance
+_prompt_optimization_job_manager = PromptOptimizationJobManager()
+
+
+# Prompt Optimization APIs
+
+
+@catch_mlflow_exception
+@_disable_if_artifacts_only
+def _optimize_prompts_handler():
+    """Handle POST /ajax-api/3.0/mlflow/optimize-prompts"""
+    request_message = _get_request_message(
+        OptimizePrompts(),
+        schema={
+            "train_dataset_id": [_assert_required, _assert_string],
+            "eval_dataset_id": [_assert_required, _assert_string],
+            "prompt_url": [_assert_required, _assert_string],
+            "scorers": [_assert_required, _assert_array],
+            "target_llm": [_assert_required, _assert_string],
+            "algorithm": [_assert_string],
+        },
+    )
+    
+    target_llm = request_message.target_llm
+    algorithm = request_message.algorithm or "DSPy/MIPROv2"
```

Comment:
I think we can propagate null and delegate the default handling to `OptimizerConfig`